src/NuGet.Core/NuGet.Commands/Utility/MSBuildProjectFrameworkUtility.cs

```diff
@@ -221,16 +298,37 @@ public static NuGetFramework GetProjectFrameworkReplacement(NuGetFramework frame
             return framework;
         }
 
-        private static string GetPropertyOrNull(string propertyName, IDictionary<string, string> projectProperties)
+        /// <summary>
+        /// Get a NuGetFramework out of the passed object. The argument is expected to either be a <see cref="NuGetFramework"/> or <see cref="string"/>.
+        /// </summary>
+        private static NuGetFramework GetAsNuGetFramework(object arg)
+        {
+            if (arg is NuGetFramework)
+            {
+                return (NuGetFramework)arg;
```

Comment:
```suggestion
            if (arg is NuGetFramework nugetFramework)
            {
                return nugetFramework;
```

I don't know if the compiler is smart enough to automatically avoid doing the typecast twice, but this makes sure.